map/transit/transit_reader.cpp

```diff
@@ -9,73 +11,30 @@
 using namespace routing;
 using namespace std;
 
+// TransitReader ----------------------------------------------------------------------------------
 bool TransitReader::Init(MwmSet::MwmId const & mwmId)
-{
-  m_src = move(GetReader(mwmId));
-  if (m_src != nullptr)
-    ReadHeader();
-  return IsValid();
-}
-
-bool TransitReader::IsValid() const
-{
-  return m_src != nullptr;
-}
-
-unique_ptr<TransitReader::TransitReaderSource> TransitReader::GetReader(MwmSet::MwmId const & mwmId)
 {
   MwmSet::MwmHandle handle = m_index.GetMwmHandleById(mwmId);
   if (!handle.IsAlive())
   {
     LOG(LWARNING, ("Can't get mwm handle for", mwmId));
-    return unique_ptr<TransitReaderSource>();
+    return {};
   }
   MwmValue const & mwmValue = *handle.GetValue<MwmValue>();
   if (!mwmValue.m_cont.IsExist(TRANSIT_FILE_TAG))
-    return unique_ptr<TransitReaderSource>();
-  FilesContainerR::TReader reader = mwmValue.m_cont.GetReader(TRANSIT_FILE_TAG);
-  return my::make_unique<TransitReaderSource>(reader);
-}
-
-void TransitReader::ReadHeader()
-{
-  try
-  {
-    transit::FixedSizeDeserializer<ReaderSource<FilesContainerR::TReader>> fixedSizeDeserializer(*m_src.get());
-    fixedSizeDeserializer(m_header);
-  }
-  catch (Reader::OpenException const & e)
-  {
-    LOG(LERROR, ("Error while reading transit header.", e.Msg()));
-    throw;
-  }
-}
-
-void TransitReader::ReadStops(vector<transit::Stop> & stops)
-{
-  ReadTable(m_header.m_stopsOffset, stops);
-}
+    return {};
 
-void TransitReader::ReadShapes(vector<transit::Shape> & shapes)
-{
-  ReadTable(m_header.m_shapesOffset, shapes);
-}
-
-void TransitReader::ReadTransfers(vector<transit::Transfer> & transfers)
-{
-  ReadTable(m_header.m_transfersOffset, transfers);
-}
-
-void TransitReader::ReadLines(vector<transit::Line> & lines)
-{
-  ReadTable(m_header.m_linesOffset, lines);
+  m_reader = make_unique<FilesContainerR::TReader>(mwmValue.m_cont.GetReader(TRANSIT_FILE_TAG));
```

Comment:
`#include "base/stl_add.hpp"`
s/make_unique/my:: make_unique

Reply:
@bykoianko , please, fix this.

Reply:
It's not good that you create a Reader from mwmValue, and mwmValue is a dangling reference, that may be freed at the end of the function.

Reply:
Done

Reply:
> It's not good that you create a Reader from mwmValue, and mwmValue is a dangling reference, that may be freed at the end of the function.

As a variant I may return shared_ptr<Reader> instead of the dangling ref. What do you think about that?

Reply:
I construct MwmHandle and use Reader at the same scope. PTAL at corresponding commit.